### PR TITLE
Oauth metrics docs

### PR DESCRIPTION
**Pull request by mstruk**
### Type of change

- Documentation

### Description

Example OAuth Metrics configurations, dashboards, and documentation 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by scholzj** [comment]
You cannot make changes to `examples/`. You have to do that to `packaging/examples`.

**Comment by scholzj** [comment]
I also think the use of OAuth is very rare. So it should have a separate dashboard and examples with the config maps or have it commented out or something. Because for most users this will be useless.

**Comment by mstruk** [comment]
Ok, the exmples vs. packaging/examples thing is detected by the build. I'll fix it.

For Grafana dashboard I'll create `strimzi-kafka-oauth.json` with same basic structure but with only the new widgets. 
The prometheus exporter mappings in metrics yaml files ... I can create duplicates containing the additional `strimzi.oauth` mappings. I can also move them under the `security/oauth/metrics` for example. 

My concern there is to try and make as easy as possible for users to activate all the necessary mappings to export all the metrics. I think for promethes mapping yamls people just copy paste the whole file content, and that's it. So it's easier if the number of these files is minimal, and each covers the whole gamut of possible metrics the user might be interested in.

**Comment by im-konge** [comment]
@mstruk I already added those Oauth metrics into `packaging/examples/metrics/oauth-metrics.yaml` so anyone interested in using those metrics needs to copy and paste it to other examples.

**Comment by mstruk** [comment]
Ok, I guess we need some docs then to instruct people  that they have to manually compose their own metrics if they want to see OAuth metrics. Additionally it would make sense to have a ready to go example where no manual composition is needed. I would add that under `packaging/examples/security/oauth`.

File: documentation/assemblies/metrics/assembly-metrics.adoc [hunk 1]
```diff
@@ -11,6 +11,9 @@ You can use Prometheus and Grafana to monitor your Strimzi deployment.
 You can monitor your Strimzi deployment by viewing key metrics on dashboards and setting up alerts that trigger under certain conditions.
 Metrics are available for each of the components of Strimzi.
 
+Some metrics have to be explicitly enabled to be collected. For example, Kafka CR 'spec.kafka.listeners.authentication' and 'spec.kafka.authorization' have 'enableMetrics' property which needs to be set to 'true' when using 'oauth' authentication or 'keycloak' authorization for corresponding metrics to be collected.
+Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.
```
Comment (scholzj) [inline documentation/assemblies/metrics/assembly-metrics.adoc:15]: I think this is misleading as it suggests this applies in general. While in reality, it applies only to `oauth` authentication and `opa` authorization / `keycloak` authentication. So I think it needs to be approached from the other side ... i.e. _if you are using one of these, you will need to explicitly enable the metrics ..._

**Review by scholzj — COMMENTED** [review]
I think this looks mostly good. But you still modify the `examples/`, so that is why the vbuild is failing.

File: documentation/assemblies/metrics/assembly-metrics.adoc (hunk 1 above)
Comment (mstruk) [reply documentation/assemblies/metrics/assembly-metrics.adoc:15]: Ok, how about:

```suggestion
If you are using `oauth` authentication or `opa` / `keycloak` authorization the collection of their specific metrics has to be explicitly enabled. For example, Kafka CR 'spec.kafka.listeners.authentication' and 'spec.kafka.authorization' have 'enableMetrics' property which needs to be set to 'true'.
Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.
```

Comment (scholzj) [reply documentation/assemblies/metrics/assembly-metrics.adoc:15]: Yeah, I think this sounds good to me. It makes it more clear who should worry about this.

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
@mstruk Can you please look at the DCO sign-off issue?

**Comment by mstruk** [comment]
DCO fixed.

**Review by im-konge — APPROVED** [review]

File: documentation/assemblies/metrics/assembly-metrics.adoc [hunk 2]
```diff
@@ -11,6 +11,9 @@ You can use Prometheus and Grafana to monitor your Strimzi deployment.
 You can monitor your Strimzi deployment by viewing key metrics on dashboards and setting up alerts that trigger under certain conditions.
 Metrics are available for each of the components of Strimzi.
 
+If you are using `oauth` authentication or `opa` / `keycloak` authorization the collection of their specific metrics has to be explicitly enabled. For example, Kafka CR 'spec.kafka.listeners.authentication' and 'spec.kafka.authorization' have 'enableMetrics' property which needs to be set to 'true'.
```
Comment (PaulRMellor) [inline documentation/assemblies/metrics/assembly-metrics.adoc:14]:
```suggestion
You can also collect metrics specific to `oauth` authentication and `opa` or `keycloak` authorization. 
You do this by setting the `enableMetrics' property to `true` in the listener configuration of the `Kafka` resource. 
For example, set `enableMetrics'  to `true` in `spec.kafka.listeners.authentication` and `spec.kafka.authorization`.
```

File: documentation/assemblies/metrics/assembly-metrics.adoc [hunk 3]
```diff
@@ -11,6 +11,9 @@ You can use Prometheus and Grafana to monitor your Strimzi deployment.
...
+Similarly for client authentication in KafkaBridge, KafkaConnect, KafkaMirrorMaker, or KafkaMirrorMaker2 CRs when 'oauth' type is used.
```
Comment (PaulRMellor) [inline documentation/assemblies/metrics/assembly-metrics.adoc:15]:
```suggestion
Similarly, you can enable metrics for `oauth` authentication in the `KafkaBridge`, `KafkaConnect`, `KafkaMirrorMaker`, and `KafkaMirrorMaker2` custom resources.  
```

File: packaging/examples/security/keycloak-authorization/README.md [hunk 4]
```diff
@@ -14,5 +14,8 @@ Full instructions for the example are available in the [Strimzi Documentation](h
 - [kafka-ephemeral-oauth-single-keycloak-authz.yaml](./kafka-ephemeral-oauth-single-keycloak-authz.yaml)
   - The Kafka CR that defines a single-node Kafka cluster with `oauth` authentication and `keycloak` authorization,
     using the `kafka-authz` realm. See [full example instructions](https://strimzi.io/docs/operators/in-development/configuring.html#proc-oauth-authorization-keycloak-example_str) for proper preparation and deployment.
+- [kafka-ephemeral-oauth-single-keycloak-authz-metrics.yaml](./kafka-ephemeral-oauth-single-keycloak-authz-metrics.yaml)
```
Comment (PaulRMellor) [inline packaging/examples/security/keycloak-authorization/README.md:17]: Do we want to say something about this in this section: https://strimzi.io/docs/operators/in-development/deploying.html#example_files_provided_with_strimzi

**Review by PaulRMellor — COMMENTED** [review]
Looking good. I made a couple of suggestion for the doc. Also wondering whether we need to mention the new example file in the deploying guide.

File: documentation/assemblies/metrics/assembly-metrics.adoc (hunk 2 above)
Comment (mstruk) [reply documentation/assemblies/metrics/assembly-metrics.adoc:14]: Maybe to be consistent:

You do this by setting the `enableMetrics' property to `true` in the listener configuration or authorization configuration of the `Kafka` resource.

File: packaging/examples/security/keycloak-authorization/README.md (hunk 4 above)
Comment (mstruk) [reply packaging/examples/security/keycloak-authorization/README.md:17]: You mean add something like:

The `keycloak-authorization` directory also contains an example with enabled metrics for `oauth` authentication and `keycloak` authorization.

File: documentation/assemblies/metrics/assembly-metrics.adoc [hunk 5]
```diff
@@ -11,6 +11,11 @@ You can use Prometheus and Grafana to monitor your Strimzi deployment.
 You can monitor your Strimzi deployment by viewing key metrics on dashboards and setting up alerts that trigger under certain conditions.
 Metrics are available for each of the components of Strimzi.
 
+You can also collect metrics specific to `oauth` authentication and `opa` or `keycloak` authorization. 
+You do this by setting the `enableMetrics' property to `true` in the listener configuration of the `Kafka` resource.
```
Comment (mstruk) [inline documentation/assemblies/metrics/assembly-metrics.adoc:15]:
```suggestion
You do this by setting the `enableMetrics` property to `true` in the listener configuration or authorization configuration of the `Kafka` resource.
```

Comment (mstruk) [reply documentation/assemblies/metrics/assembly-metrics.adoc:15]: @PaulRMellor Also fixed a backtick there.

**Comment by scholzj** [comment]
@PaulRMellor Do you have something more for this PR?

**Comment by PaulRMellor** [comment]
> @PaulRMellor Do you have something more for this PR?

a couple of backticks comments above, which haven't been fixed yet

**Review by PaulRMellor — APPROVED** [review]

**Pull request merged**